tests/snapshots/tests.contrib.rq.test_rq.test_enqueue.snap

```diff
@@ -0,0 +1,59 @@
+[[{"name" "rq.queue.enqueue_job"
+   "service" "rq"
+   "resource" "tests.contrib.rq.jobs.job_add1"
+   "type" "worker"
+   "error" 0
+   "span_id" 0
+   "trace_id" 0
+   "parent_id" nil
+   "start" 1629316603180780000
+   "duration" 9730000
+   "meta" {"runtime-id" "fce7c109fd434f178c779045713e12fe"
+           "queue.name" "q"
+           "job.id" "6fcf262c-a9ff-4056-bd48-0bbca86151f7"
+           "job.func_name" "tests.contrib.rq.jobs.job_add1"}
+   "metrics" {"_dd.agent_psr" 1.0
+              "_sampling_priority_v1" 1
+              "system.pid" 80738
+              "_dd.tracer_kr" 1.0}}]
+ [{"name" "rq.worker.perform_job"
+   "service" "rq-worker"
+   "resource" "tests.contrib.rq.jobs.job_add1"
+   "type" "worker"
+   "error" 0
+   "span_id" 0
+   "trace_id" 1
+   "parent_id" nil
+   "start" 1629316605903245000
+   "duration" 16484000
+   "meta" {"runtime-id" "8aa0e907db5e405688b506044514fd0f"
+           "job.id" "6fcf262c-a9ff-4056-bd48-0bbca86151f7"}
+   "metrics" {"_dd.agent_psr" 1.0
+              "_sampling_priority_v1" 1
+              "system.pid" 80759
+              "_dd.tracer_kr" 1.0}}
+      {"name" "rq.job.perform"
+       "service" "rq-worker"
+       "resource" "tests.contrib.rq.jobs.job_add1"
+       "error" 0
+       "span_id" 1
+       "trace_id" 1
+       "parent_id" 0
+       "start" 1629316605909347000
+       "duration" 4910000
+       "meta" {"job.id" "6fcf262c-a9ff-4056-bd48-0bbca86151f7"}}]
+ [{"name" "rq.job.fetch"
+   "service" "rq-worker"
+   "resource" "rq.job.fetch"
+   "error" 0
+   "span_id" 0
+   "trace_id" 2
+   "parent_id" nil
+   "start" 1629316605892859000
```

Comment:
This start time is before `rq.worker.perform_job` which makes sense but I would have expected a span that was the parent of both here as they should be done by the worker instance in sequence.

Looking at the source code, it appears that a job is fetched and then handed off for execution within a work loop.

https://github.com/rq/rq/blob/b80045d615d539cf4c565c5541694efe13ea648e/rq/worker.py#L599
https://github.com/rq/rq/blob/b80045d615d539cf4c565c5541694efe13ea648e/rq/worker.py#L607

Being that this work is connected, these two spans should share a common parent. A possibility here could be to have a parent `work` span where the first child is the fetch. The parent span can be left open after the fetch is complete, stashed in the job object, and then closed once the job has executed.

Reply:
would there be any value in measuring the batches of jobs on the worker side? It would expose the usage patterns of the worker maybe but I'm not sure if that's useful to a user.